### DISCUSSION TITLE
How do I host multiple subdomains from one container?

### DISCUSSION THREAD

**Discussion by tzuleger** [nginx-proxy/nginx-proxy#2509] [Q&A]
I am trying to use [OvenMediaEngine](https://airensoft.gitbook.io/) which will have a single Docker Container. This container has multiple web interfaces that I want to use, `3334` for HLS/LL-HLS and `80` for the API.

I tried using [VIRTUAL_HOST_MULTIPORTS](https://github.com/nginx-proxy/nginx-proxy/tree/main/docs#virtual-ports) but my sites keep returning `503` with nginx's default "Service Unavailable" page.

These are the commands I run in a bash script:

```
sudo docker rm -f oven-media-engine

VIRTUAL_HOST_MULTIPORTS_VALUE="{\"api.ome.example.com\":{\"/\":{\"port\":80,\"dest\":\"\"}},\"live.ome.example.com\":{\"/\":{\"port\":3334,\"dest\":\"\"}}}"
HOST_NAMES=api.ome.example.com,live.ome.example.com
HOST_IP=$(curl ipv4.icanhazip.com)

docker run -d --name oven-media-engine \
        -v ~/OvenMediaEngine/mounts/logs/:/var/log/ovenmediaengine \
        -v ~/OvenMediaEngine/mounts/ome-origin-conf/:/opt/ovenmediaengine/bin/origin_conf \
        -v ~/OvenMediaEngine/mounts/ome-edge-conf/:/opt/ovenmediaengine/bin/edge_conf \
        -v ~/nginx/sslconf/certs/:/cert/ \
        --network nginx-net \
        -h $HOST_NAMES \
        -e LETSENCRYPT_HOST=$HOST_NAMES \
        -e VIRTUAL_HOST_MULTIPORTS=$VIRTUAL_HOST_MULTIPORTS_VALUE \
        -e OME_HOST_IP=$HOST_IP \
        -p 1935:1935 \
        -p 9999:9999/udp \
        -p 9000:9000 \
        -p 3333:3333 \
        -p 3334:3334 \
        -p 3478:3478 \
        -p 10000-10009:10000-10009/udp \
        airensoft/ovenmediaengine:latest
```

Am I doing this wrong? The only examples I see are with docker-compose, so I figured maybe this is done differently by the CLI command.

**Comment by stefandesu** [comment]
I think this was a bug! I had the same issue when trying out `VIRTUAL_HOST_MULTIPORTS` today where I only got 503s even with the simplest configuration, and it only started working after I updated our nginx-proxy Docker image. I couldn't find any reference to this in the release notes, but you should update and try again.

**Reply by buchdag** [reply]
In addition to the above (`VIRTUAL_HOST_MULTIPORTS` was added in 1.6.0, just over 4 months ago), I'd check the following:
- wrap your shell variables in double quote when accessing them, it will never hurt: `VIRTUAL_HOST_MULTIPORTS="$VIRTUAL_HOST_MULTIPORTS_VALUE"`
- don't use escaped double quote unless you explicitly need them as they make your JSON really hard to parse, wrapping your whole bash variable in single quote is okay, it's just a string.
- start simple, with only the API, without certificates and without extra unneeded configuration: `VIRTUAL_HOST_MULTIPORTS_VALUE='{"api.ome.example.com": {"/": {} } }'`
- if it work add the other port, if it does not inspect the generated config with `docker exec yourProxyContainer nginx -T`. If docker-gen fail to parse the variable's content as YAML / JSON, there will be warning as a comment about this in the config.
